Engine/DataFeeds/Enumerators/FillForwardEnumerator.cs

```diff
@@ -293,8 +298,7 @@ protected virtual bool RequiresFillForwardData(TimeSpan fillForwardResolution, B
                 var potentialUtc = _offsetProvider.ConvertToUtc(item.ReferenceDateTime) + item.Interval;
                 var potentialInTimeZone = _offsetProvider.ConvertFromUtc(potentialUtc);
                 var potentialBarEndTime = RoundDown(potentialInTimeZone, item.Interval);
-                // apply the same timezone to next and potential bars
-                // calculate next endTime by adding duration to the previous data time because incoming next.EndTime is swallowing one hour
+                // apply the same timezone to next and potential bars because incoming next.EndTime can swallow one hour
                 var nextEndTime = next.EndTime - daylightMovement;
                 if (potentialBarEndTime < nextEndTime)
```

Comment:
Just a note: `potentialBarEndTime` in the other hand isn't affected by the daylight savings since it's generated from UTC after fixing a similar issue that caused an infinite loop situation see https://github.com/QuantConnect/Lean/issues/3707, and before this change `next.EndTime` was affected and it caused this issue

Reply:
exactly. potentialBarEndTime is calculated using right _offsetProvider movement and didn't swallow hour anywhere